### PR TITLE
ci: Rework CI based on current XCbuild command

### PR DESCRIPTION
**Pull request by adrien-coye**
### Abstract

Given the new test target for API tests, I'm making discrete test targets for UT / UI / API tests and we should be able to quickly see what's wrong when a test fails.

The use of dedicated CI Checks makes it easier to read and parallelise better.

### Misc

Superseding https://github.com/Infomaniak/ios-kDrive/pull/1090

~~Still in draft as needs minor rework to work with this PR that should be merged first : https://github.com/Infomaniak/ios-kDrive/pull/1092~~

### CODE REVIEW

**Comment by adrien-coye** [comment]
Right now the project is correctly setup in this PR, but API tests and UI tests are failing.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,7 +26,111 @@ jobs:
         run: |
           touch $ENV_PATH
           echo -e "enum Env {\n    static let token = \"$ACCESS_TOKEN\"\n\n    static let driveId = 420132\n\n    static let userId = $USER_ID\n\n    static let inviteUserId = $INVITE_USER_ID\n\n    static let inviteMail = \"$INVITE_EMAIL\"\n\n    static let inviteTeam = 0\n\n    static let commonDocumentsId = 3\n}" > $ENV_PATH
+      - name: Tuist generate
+        run: .tuist-bin/tuist generate
       - name: Build
-        run: .tuist-bin/tuist build --generate kDrive
+        run: .tuist-bin/tuist build kDrive
+
+  test_unit_itegration:
```
Comment (valentinperignon) [inline .github/workflows/ci.yml:34]:
```suggestion
  test_unit_integration:
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,7 +26,111 @@ jobs:
...
+        run: |
+          touch $ENV_PATH
+          echo -e "enum Env {\n    static let token = \"$ACCESS_TOKEN\"\n\n    static let driveId = 420132\n\n    static let userId = $USER_ID\n\n    static let inviteUserId = $INVITE_USER_ID\n\n    static let inviteMail = \"$INVITE_EMAIL\"\n\n    static let inviteTeam = 0\n\n    static let commonDocumentsId = 3\n}" > $ENV_PATH
+      - name: Tuist generate
+        run: .tuist-bin/tuist generate
       - name: Test
-        run: xcodebuild test -workspace kDrive.xcworkspace -scheme kDrive -destination "platform=iOS Simulator,name=iPhone 13" -only-testing:kDriveTests/DriveApiTests
+        run: xcodebuild test -workspace kDrive.xcworkspace -scheme kDrive -allowProvisioningUpdates -only-testing "kDriveUITests" -destination "platform=iOS Simulator,name=iPhone 15,OS=17.0"
+
+  test_unit_itegration_legacy:
```
Comment (adrien-coye) [inline .github/workflows/ci.yml:86]:
```suggestion
  test_unit_integration_legacy:
```

**Comment by adrien-coye** [comment]
superseeded by https://github.com/Infomaniak/ios-kDrive/pull/1132

**Pull request closed without merging**